### PR TITLE
Use default dtype in `L.NegativeSampling`

### PR DESCRIPTION
**Pull request by takagi**
This PR follows #4582, making `L.NegativeSampling' support the default dtype.

### CODE REVIEW

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/links/loss/negative_sampling.py [hunk 1]
```diff
@@ -29,11 +31,12 @@ class NegativeSampling(link.Link):
...
     """
 
-    def __init__(self, in_size, counts, sample_size, power=0.75):
+    def __init__(self, in_size, counts, sample_size, power=0.75, dtype=None):
         super(NegativeSampling, self).__init__()
+        dtype = chainer.get_dtype(dtype)
         vocab_size = len(counts)
         self.sample_size = sample_size
-        power = numpy.float32(power)
+        power = dtype.type(power)
```
Comment (toslunar) [inline chainer/links/loss/negative_sampling.py]: `power.dtype` in the next line could be fixed, while it's not necessarily related to the PR.

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/links/loss/negative_sampling.py (hunk 1 above)
Comment (takagi) [reply chainer/links/loss/negative_sampling.py]: I fixed!

Comment (toslunar) [reply chainer/links/loss/negative_sampling.py]: Sorry.  I meant “... could be fixed, too”.

Comment (takagi) [reply chainer/links/loss/negative_sampling.py]: Just as follows?
```Python
power = dtype.type(power)
p = numpy.array(counts, dtype)
```

Comment (toslunar) [reply chainer/links/loss/negative_sampling.py]: Yes!

Comment (toslunar) [reply chainer/links/loss/negative_sampling.py]: Thanks!!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
I've rebased on the current master to follow #5787 to pass CI on Jenkins.

**Comment by takagi** [comment]
Jenkins, test this please.

**Pull request merged**